### PR TITLE
[mesheryctl] improve the error messages shown to users for "mesheryctl"

### LINKED ISSUE #6265: [mesheryctl] improve the error messages shown to users for "mesheryctl"
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._ For additional context, see https://github.com/meshery/meshery/pull/5661.

#### Current Behavior
Responses from `mesheryctl` commands don't always have the best user experience in mind. 

#### Desired Behavior
Improve the error messages in mesheryctl commands from the user's perspective. 
1. Ensure that there are proper error messages when users:
    1. Give more arguments to a command than required
    1. Misspell a command
    1. Do not give the required input via flags or args
    1. Use the command without any arguments or flags.
1. The error messages being displayed to users are meaningful from user's perspective. And if applicable, a potential fix for the problem should be shown too.  For example
    1. "failed to unmarshal response body" could be modified to "couldn't process JSON response from server"
    1. "manifest not found" could be modified to "manifest not found. Please provide the path to the app manifest."

#### Implementation
The commands in scope of this issue are:

- [ ] `mesheryctl` 
- [ ] `mesheryctl version`
- [ ] `mesheryctl completion`

#### Acceptance Tests
1. Contributor adhered to the [Meshery CLI Style Guide](https://docs.meshery.io/project/contributing/contributing-cli#meshery-cli-style-guide).

 --- 
  #### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)
  📚 For the **Reference** head over to this [pr](https://github.com/meshery/meshery/pull/5661).

### PR DESCRIPTION
**Pull request by XDRAGON2002**
Signed-off-by: Anant Vijay <anantvijay3@gmail.com>

**Description**

This PR fixes #6265

**Notes for Reviewers**
The error messages have been improved and have been made more user friendly

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/completion.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ var completionCmd = &cobra.Command{
 		case "fish":
 			return cmd.Root().GenFishCompletion(os.Stdout, true)
 		default:
-			return errors.New("shell not supported")
+			return errors.New("shell not supported, requires one of [bash|zsh|fish]")
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/completion.go:80]: Could we make it more meaningful? Like....

```suggestion
			return errors.New("shell not supported, requires one of the shells eg:[bash|zsh|fish]")
```

// @harkiratsm @leecalcote

**Review by alphaX86 — CHANGES_REQUESTED** [review]
PR reviewed. Do the following changes...

Oh and you need to fix the lint issues too @XDRAGON2002

**Review by hershd23 — COMMENTED** [review]
Mostly LGTM. Don't really have anything to say about the changes. Do fix the linting issues as suggested earlier

**Comment by XDRAGON2002** [comment]
> Mostly LGTM. Don't really have anything to say about the changes. Do fix the linting issues as suggested earlier

@hershd23 The linting checks are broken at the moment, see #6385.

File: mesheryctl/internal/cli/root/error.go [hunk 2]
```diff
@@ -37,9 +37,9 @@ func ErrGettingRequestContext(err error) error {
 }
 
 func ErrInvalidAPIResponse(err error) error {
-	return errors.New(ErrInvalidAPIResponseCode, errors.Fatal, []string{"Invalid API response"}, []string{"Invalid API response", err.Error()}, []string{}, []string{})
+	return errors.New(ErrInvalidAPIResponseCode, errors.Fatal, []string{"Invalid API response encountered"}, []string{"Invalid API response encountered", err.Error()}, []string{}, []string{})
 }
 
 func ErrUnmarshallingAPIData(err error) error {
-	return errors.New(ErrUnmarshallingAPIDataCode, errors.Fatal, []string{"Error unmarshalling API data"}, []string{"Error unmarshalling API data", err.Error()}, []string{}, []string{})
+	return errors.New(ErrUnmarshallingAPIDataCode, errors.Fatal, []string{"Error processing json API data"}, []string{"Error processing json API data", err.Error()}, []string{}, []string{})
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/error.go:44]: More friendly human language. 👍

**Review by leecalcote — APPROVED** [review]
LGTM

**Comment by legendarykamal** [comment]
Hi @leecalcote, good to merge ? let's do it : or not 

`(ternary operator use ) `

**Comment by leecalcote** [comment]
Ha! 🙂

**Pull request merged**